providers/implementations/macs/hmac_prov.c

```diff
@@ -190,12 +263,25 @@ static int hmac_set_ctx_params(void *vmacctx, const OSSL_PARAM params[])
         if (p->data_type != OSSL_PARAM_OCTET_STRING)
             return 0;
 
+        if (macctx->keylen > 0)
+            OPENSSL_secure_clear_free(macctx->key, macctx->keylen);
+        /* Keep a copy of the key if we need it for TLS HMAC */
```

Comment:
It's shame that we have to always keep the key copy even if it is not needed (with regular HMAC).

Reply:
Yes, this is a shame. A possible solution is to set yet another parameter to indicate that we're going to do TLS-HMAC. But this potentially becomes quite brittle since the order that you set parameters becomes important. Also I didn't want to add the extra complexity to the API, so I didn't go down that route.